tensorflow_recommenders_addons/dynamic_embedding/python/keras/callbacks.py

```diff
@@ -106,6 +107,10 @@ def __init__(self, root_rank=0, device='', local_variables=None):
       self.register_local_var(var)
 
 
+@deprecated(
```

Comment:
Is the warning always triggered? It's recommended to show it only when the users actually refer to the `AllToAllEmbedding`.

Reply:
Only show when this class was called __init__ or __new__.

```python
def deprecated_wrapper(func_or_class):
    """Deprecation wrapper."""
    if isinstance(func_or_class, type):
      # If a class is deprecated, you actually want to wrap the constructor.
      cls = func_or_class
      if cls.__new__ is object.__new__:
        # If a class defaults to its parent's constructor, wrap that instead.
        func = cls.__init__
        constructor_name = '__init__'
        decorators, _ = tf_decorator.unwrap(func)
        for decorator in decorators:
          if decorator.decorator_name == 'deprecated':
            # If the parent is already deprecated, there's nothing to do.
            return cls
      else:
        func = cls.__new__
        constructor_name = '__new__'

    else:
      cls = None
      constructor_name = None
      func = func_or_class
```